.github/workflows/ci.yml

```
@@ -39,63 +85,95 @@ jobs:
       - name: Disk space cleanup
         uses: jlumbroso/free-disk-space@54081f138730dfa15788a46383842cd2f914a1be # v1.3.1
 
-      - name: Run pipeline with test data
+      - name: Install nf-test
         run: |
-          nextflow run ${GITHUB_WORKSPACE} -profile test,docker --outdir ./results
-
-  profile:
-    name: Run additional profile tests
-    # Only run on push if this is the nf-core dev branch (merged PRs)
-    if: ${{ github.event_name != 'push' || (github.event_name == 'push' && github.repository == 'nf-core/metapep') }}
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        tests: [
-            "test_model_information",
-            "test_mouse_all_pep_lengths",
-            "test_assembly_only",
-            "test_bins_only",
-            "test_coassembly",
-            "test_mhcflurry",
-            "test_mhcnuggets_1",
-            "test_mhcnuggets_2",
-          ] # add further test profiles here, will be run in parallel (but only with one nextflow version)
-    steps:
-      - name: Check out pipeline code
-        uses: actions/checkout@v2
+          wget -qO- https://code.askimed.com/install/nf-test | bash -s $NFTEST_VER
+          sudo mv nf-test /usr/local/bin/
 
-      - name: Install Nextflow
+      - name: Run nf-test
         run: |
-          wget -qO- get.nextflow.io | bash
-          sudo mv nextflow /usr/local/bin/
+          nf-test test --tag ${{ matrix.tags }} --profile ${{ matrix.tags }},${{ matrix.profile }} --junitxml=test.xml
 
-      - name: Run pipeline with profile ${{ matrix.tests }}
+      - name: Output log on failure
+        if: failure()
         run: |
-          nextflow run ${GITHUB_WORKSPACE} -profile ${{ matrix.tests }},docker  --outdir ./results
+          sudo apt install bat > /dev/null
+          batcat --decorations=always --color=always ${{ github.workspace }}/.nf-test/tests/*/output/pipeline_info/software_versions.yml
 
-  all_profiles:
-    name: Run all additional profile tests that require ncbi credentials
+      - name: Publish Test Report
+        uses: mikepenz/action-junit-report@v3
+        if: always() # always run even if the previous step fails
+        with:
+          report_paths: "*.xml"
+
+  profile_download:
```

```suggestion
  test_profile_download:
```
Could this be named consistently?